cvmfs/sync_mediator.h

```diff
@@ -50,6 +51,49 @@ struct Counters;
 
 namespace publish {
 
+class SyncDiffReporter : public DiffListener {
+ public:
+  explicit SyncDiffReporter(bool print_dots,
```

Comment:
Instead of a the `print_dots` bool, let's define an `enum` with the different print modes: `kPrintDots`, `kPrintChanges`